### DISCUSSION TITLE
Request API in an other file of +Layout.tsx

### DISCUSSION THREAD

**Discussion by Nelie-Taylor** [vikejs/vike#1670] [Discussion] [answered]
Hi.
Thank you for greate framework.
![image](https://github.com/vikejs/vike/assets/12018657/ddf74580-f69f-4f22-9e8d-3354a2453abc)
I have a case:
- In +Layout.tsx load info from server. 
- When i access to page project, +data.tsx not call when i access to project page.

Please help me load data in +date.tsx or how to load data from other page when include +Layout.tsx.
Thank you so much.

**Comment by phonzammi** [comment]
Hi @Nelie-Taylor, are you using [vike-react](https://github.com/vikejs/vike-react)?

**Reply by phonzammi** [reply]
if not, have you tried this https://vike.dev/useData#without-vike-react-vue-solid ?

**Reply by Nelie-Taylor** [reply]
Hi @phonzammi, thank you for reply, i use react typescript.

**Reply by phonzammi** [reply]
That doesn't answer my question

**Reply by Nelie-Taylor** [reply]
Hi @phonzammi.
My problem is data of ```/pages/+data.js``` is overwritten by ```/pages/project/+data.ts```.

**Comment by brillout** [comment]
(Your issue seems to be a user-land issue, but feel free to let me know and provide a minimal reproduction if you think it's a Vike issue.)

**Reply by Nelie-Taylor** [reply]
Hi, thank you for replay.
This is my page constructs.
![image](https://github.com/vikejs/vike/assets/12018657/1120f051-0c60-4a14-8acf-f1367fe1e75c)
```/pages/+data.ts``` is request data from server to show in ```+Layout.tsx```
![image](https://github.com/vikejs/vike/assets/12018657/70acb7d0-60fd-428d-ae42-2553c752e465)
I got data from ```+data.ts```
![image](https://github.com/vikejs/vike/assets/12018657/b04e54a6-fa62-4c47-8d62-b5aee35f8e50)
But when i access to ```/project``` (```pages/project/+Page.tsx```). Data of ```/pages/+data.js``` is overwritten by ```/pages/project/+data.ts```.
How to keep data of ```/pages/+data.js``` in ```/pages/project/+data.ts```.
thank you.

**Comment by Nelie-Taylor** [comment, marked as answer]
Hi, i resolve problem with update file ```/rerender/+onRenderHtml.tsx``` request data from server and retrun in pageContext
![image](https://github.com/vikejs/vike/assets/12018657/caa0155b-76af-4612-a643-6babd6ddd845)
